include/pf-applications/grain_tracker/representation.h

```diff
@@ -0,0 +1,111 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2024 by the hpsint authors
+//
+// This file is part of the hpsint library.
+//
+// The hpsint library is free software; you can use it, redistribute
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 3.0 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE.MD at
+// the top level directory of hpsint.
+//
+// ---------------------------------------------------------------------
+
+#pragma once
+
+#include <deal.II/base/point.h>
+
+#include <pf-applications/base/debug.h>
+
+#include <boost/serialization/export.hpp>
+#include <boost/serialization/serialization.hpp>
+
+namespace GrainTracker
+{
+  using namespace dealii;
+
+  // This base erases type
+  struct Representation
+  {
+    virtual double
+    distance(const Representation &other) const = 0;
+
+    virtual void
+    print(std::ostream &stream) const = 0;
+
+    virtual std::unique_ptr<Representation>
+    clone() const = 0;
+
+    virtual ~Representation()
+    {}
+
+    template <class Archive>
+    void
+    serialize(Archive &ar, const unsigned int /*version*/)
+    {
+      (void)ar;
+    }
```

Comment:
This is a bit suboptimal, since the function cannot be made virtual.

Reply:
And there is no need to do it in fact for the boost serialization to work. This function is required here for polymorphic serialization to work. You may see that I have a unique pointer to `Representation` inside `Segment`, but serialization needs to be able to invoke `serialize` the most derived class and does it once `BOOST_CLASS_EXPORT` macro is applied to the serializable classes which are inherited from a given base.